src/raster/r.suitability.regions/r.suitability.regions.py

```diff
@@ -0,0 +1,410 @@
+#% description: Compute map with average suitability per region
+#% guisection: Reporting stats
+#%end
+#%flag
+#% key: a
+#% label: Calculates area of clumped areas (hectares)
+#% description: Calculates area of clumped areas
+#% guisection: Reporting stats
+#%end
+
+#%flag
+#% key: k
+#% label: Keep map with all suitable cells
+#% description: Keep map that show suitable areas (irrespective of clump size)
+#% guisection: Optional
+#%end
+
+#%flag
+#% key: f
+#% label: Keep suitability based on focal statistics
+#% description: Keep suitabiliyt based on focal statistics
+#% guisection: Optional
+#%end
+
+import sys
+import os
+import atexit
+import uuid
+import grass.script as gs
+
+RECLASSRULES1 = """\
+1:filled gaps
+2:suitable areas
+"""
+COLORRULES1 = """
+1 230:230:230
+""".strip()
+
+COLORRULES2 = """
+1 241:241:114
+2 139:205:85
+""".strip()
+
+
+CLEAN_LAY = []
+
+
+def cleanup():
+    """Remove temporary maps specified in the global list"""
+    cleanrast = list(reversed(CLEAN_LAY))
+    for rast in cleanrast:
+        ffile = gs.find_file(name=rast, element="cell", mapset=gs.gisenv()["MAPSET"])
+        if ffile["file"]:
+            gs.run_command("g.remove", flags="f", type="raster", name=rast, quiet=True)
+
+
+def tmpname(prefix):
+    """Generate a tmp name which contains prefix
+    Store the name in the global list.
+    Use only for raster maps.
+    """
+    tmpf = prefix + str(uuid.uuid4())
+    tmpf = tmpf.replace("-", "_")
+    CLEAN_LAY.append(tmpf)
+    return tmpf
+
+
+def tmpmask(input, absmin):
+    """Create tmp mask"""
+    rules = "*:{}:1".format(absmin)
+    tmprecode = tmpname("tmprecode")
+    gs.write_command(
+        "r.recode", input=input, output=tmprecode, rule="-", stdin=rules, quiet=True
+    )
+    return tmprecode
+
+
+def main(options, flags):
+
+    suitrast = options["suitrast"]
+    absminsuit = options["absminsuit"]
+    radius = int(options["radius"])
+    if (radius % 2) == 0:
+        gs.error("Radius should be an odd positive number")
+    focalstat = options["focalstat"]
+    minsuit = float(options["minsuit"])
+    minfrag = float(options["minfrag"])
+    maxgap = float(options["maxgap"])
+    suitreg = options["suitreg"]
+
+    # Flags
+    flag_c = flags["c"]
+    if flag_c:
+        c = "c"
+    else:
+        c = ""
+    flag_d = flags["d"]
+    if flag_d:
+        d = "d"
+    else:
+        d = ""
+    flag_z = flags["z"]
+    flag_k = flags["k"]
+    flag_f = flags["f"]
+    flag_a = flags["a"]
+
+    # Temporary layer names
+    tmp01 = tmpname("tmp01")
+    tmp02 = tmpname("tmp02")
+    tmp02a = tmpname("tmp02a")
+    tmp03 = tmpname("tmp03")
+    tmp04 = tmpname("tmp04")
+    tmp05 = tmpname("tmp05")
+    tmp06 = tmpname("tmp06")
+    tmp07a = tmpname("tmp07a")
+    tmp07 = tmpname("tmp07")
+    tmp08 = tmpname("tmp08")
+    xtralayer = "{}_filledgaps".format(suitreg)
+
+    # Compute neighborhood statistic
+    if (radius > 1) & (len(absminsuit) == 0):
```

Comment:
Logical AND in Python is `and`. Most people would leave out the `()` around `and` operands.